src/main/kotlin/net/atos/zac/zaken/ZakenService.kt

```diff
@@ -12,27 +14,45 @@ import net.atos.client.zgw.zrc.model.RolOrganisatorischeEenheid
 import net.atos.client.zgw.zrc.model.Zaak
 import net.atos.client.zgw.ztc.ZTCClientService
 import net.atos.client.zgw.ztc.model.generated.RolType
+import net.atos.zac.event.EventingService
 import net.atos.zac.identity.model.Group
 import net.atos.zac.identity.model.User
+import net.atos.zac.websocket.event.ScreenEventType
 import net.atos.zac.zoeken.IndexeerService
 import net.atos.zac.zoeken.model.index.ZoekObjectType
-import java.util.UUID
+import java.util.*
+import java.util.logging.Logger
 
 class ZakenService @Inject constructor(
     private val zrcClientService: ZRCClientService,
     private val ztcClientService: ZTCClientService,
-    private val indexeerService: IndexeerService
+    private val indexeerService: IndexeerService,
+    private var eventingService: EventingService
 ) {
     companion object {
         private val defaultCoroutineScope = CoroutineScope(Dispatchers.Default)
+        private val LOG = Logger.getLogger(ZakenService::class.java.name)
     }
 
     /**
      * Asynchronously assigns a list of zaken to a group and/or user and updates the search index on the fly.
      */
-    fun assignZakenAsync(zaakUUIDs: List<UUID>, explanation: String? = null, group: Group? = null, user: User? = null) =
-        defaultCoroutineScope.launch {
-            zaakUUIDs.forEach { zaakUUID ->
+    @Suppress("LongParameterList")
+    fun assignZakenAsync(
+        screenEventType: ScreenEventType,
+        jobUUID: UUID,
+        zaakUUIDs: List<UUID>,
+        explanation: String? = null,
+        group: Group? = null,
+        user: User? = null
+    ) = defaultCoroutineScope.launch(CoroutineName("AssignZakenCoroutine")) {
+        LOG.fine {
+            "Started asynchronous job with UUID: $jobUUID to assign " +
+                "${zaakUUIDs.size} zaken to group and/or user"
+        }
+        val zakenAssignedList = mutableListOf<UUID>()
+        zaakUUIDs.forEach { zaakUUID ->
+            withContext(Dispatchers.IO) {
                 val zaak = zrcClientService.readZaak(zaakUUID)
```

Comment:
What happens if this call breaks? We do not send failed event, do we?

Reply:
I have not implemented error handling in the coroutine indeed. I am focussing on the happy path first. :-)

I suggest to add error handling in a subsequent PR. But I agree it needs to be added..

Reply:
But indeed, if it fails now it will not send any screen event and the exception will I think be lost in the coroutine. Also, the frontend modal will then wait forever for the screen event. Not a huge deal but we do really need error handling.